### PR TITLE
[minor] refactor Cmdliner usage

### PR DESCRIPTION
**Pull request by gasche**
The goal is to make it as simple as possible to add new fields to
configuration records (Builder.t, Server.t, etc).

The command-line-interface parts of learnocaml_main entangle the
construction of proper configuration records with unpleasant
global-mutable-variable settings. This PR splits each affected 'term'
value into smaller parts, with one purely in charge of building the
configuration record, and another dealing with the global mutation.

### CODE REVIEW

File: src/main/learnocaml_main.ml [hunk 1]
```diff
@@ -120,12 +125,14 @@ module Args = struct
         Grader_cli.dump_dot := dump_dot;
         Learnocaml_process_exercise_repository.dump_outputs := dump_outputs;
         Learnocaml_process_exercise_repository.dump_reports := dump_reports;
-        { exercises; output_json }
+        ()
```
Comment (yurug) [inline src/main/learnocaml_main.ml]: Why did you not remove this statement?

File: src/main/learnocaml_main.ml [hunk 2]
```diff
@@ -194,11 +207,13 @@ module Args = struct
         Learnocaml_process_playground_repository.playground_dir :=
           repo_dir/"playground";
         Learnocaml_process_exercise_repository.n_processes := jobs;
-        { contents_dir; try_ocaml; lessons; exercises; playground; toplevel }
+        ()
```
Comment (yurug) [inline src/main/learnocaml_main.ml]: Why did you not remove this statement?

File: src/main/learnocaml_main.ml (hunk 1 above)
Comment (gasche) [reply src/main/learnocaml_main.ml]: I guess it's a style thing, but in this particular codebase it felt natural to insist on the fact that running this term returns the `()` value (for example it explains why we match on this value in the `apply` people below). Would you like me to remove it?

**Comment by yurug** [comment]
Thank you @gasche!

**Pull request merged**